src/na/na_ofi.c

```diff
@@ -1743,47 +1742,42 @@ na_ofi_provider_check(
     rc = fi_getinfo(NA_OFI_VERSION, /* OFI version requested */
         NULL,                       /* Optional name or fabric to resolve */
         NULL,                       /* Optional service name to request */
-        0ULL,                       /* Optional flag */
+        FI_PROV_ATTR_ONLY,          /* Optional flag */
         NULL,                       /* Optional hints to filter providers */
         &providers);                /* Out: List of matching providers */
     if (rc != 0)
         return;
 
     /* look for match */
     for (cur = providers; cur; cur = cur->next) {
-        if (!strcmp(cur->fabric_attr->prov_name, na_ofi_prov_name[prov_type])) {
+        if (!strcmp(cur->fabric_attr->prov_name, na_ofi_prov_name[prov_type]) ||
+            !strcmp(
+                cur->fabric_attr->prov_name, na_ofi_prov_alt_name[prov_type])) {
```

Comment:
the flag `FI_PROV_ATTR_ONLY` returns the list of providers that are compiled in and the difference is that if a provider fails to initialize for any reason it will still be in that list.
On my machine with that patch I get:
```
psm2,psm,usnic,ofi_rxm,ofi_rxd,shm,udp,tcp,sockets,ofi_hook_perf,ofi_
```
whereas before I was getting:
```
tcp;ofi_rxm,udp;ofi_rxd,shm,udp,tcp,sockets
```
In either case the comparison with something like `verbs;ofi_rxm` would not work I think ?

Reply:
My bad, I was not reading correctly. Yes looks like we still do need the previous logic.